tools/engine_tool/lib/src/commands/build_command.dart

```diff
@@ -53,6 +53,14 @@ final class BuildCommand extends CommandBase {
       return 1;
     }
 
+    // Update dependencies if needed.
+    if (await dependenciesChanged()) {
+      final int fetchExitCode = await fetchDependencies(environment);
+      if (fetchExitCode != 0) {
+        return fetchExitCode;
```

Comment:
Is there context we could be losing in terms of why it failed, or do we just assume this won't (practically) happen?

Reply:
`fetchDependencies` will log `gclient`'s output on error, similar to how `runBuild` logs ninja output on error. I don't think there are any recoverable errors here, the user will likely need to intervene. Let me know if that doesn't address your concerns or if I misunderstood something.

Reply:
Nah that answers my question, I didn't click into it :)